### PR TITLE
Fixed 'then' and 'else' statements

### PR DESCRIPTION
**Pull request by lunixoid**
I'm a member of the Pinguem.ru competition on finding errors in open source projects. A warning, found using PVS-Studio:
fcl/include/fcl/broadphase/detail/hierarchy_tree-inl.h      1098    err     V523 The 'then' statement is equivalent to the 'else' statement.
fcl/include/fcl/broadphase/detail/hierarchy_tree-inl.h      1116    err     V523 The 'then' statement is equivalent to the 'else' statement.

### CODE REVIEW

File: include/fcl/broadphase/detail/hierarchy_tree-inl.h [hunk 1]
```diff
@@ -1094,8 +1094,7 @@ struct SelectImpl<S, AABB<S>>
     else bv.min_[0] += vel[0];
     if(vel[1] > 0) bv.max_[1] += vel[1];
     else bv.min_[1] += vel[1];
-    if(vel[2] > 0) bv.max_[2] += vel[2];
-    else bv.max_[2] += vel[2];
+    bv.max_[2] += vel[2];
```
Comment (mamoll) [inline include/fcl/broadphase/detail/hierarchy_tree-inl.h]: I suspect the right fix is this:
```
if(vel[2] > 0) bv.max_[2] += vel[2];
    else bv.min_[2] += vel[2];
```

File: include/fcl/broadphase/detail/hierarchy_tree-inl.h [hunk 2]
```diff
@@ -1112,8 +1111,7 @@ struct SelectImpl<S, AABB<S>>
...
```
Comment (mamoll) [inline include/fcl/broadphase/detail/hierarchy_tree-inl.h]: Same here.

**Comment by lunixoid** [comment]
I agree, fixed them in the next commit.

File: include/fcl/broadphase/detail/hierarchy_tree-inl.h [hunk 3]
```diff
@@ -1095,7 +1095,7 @@ struct SelectImpl<S, AABB<S>>
     if(vel[1] > 0) bv.max_[1] += vel[1];
     else bv.min_[1] += vel[1];
     if(vel[2] > 0) bv.max_[2] += vel[2];
-    else bv.max_[2] += vel[2];
+      else bv.min_[2] += vel[2];
```
Comment (jslee02) [inline include/fcl/broadphase/detail/hierarchy_tree-inl.h]: I believe @mamoll actually meant to change `bv.max_[2]` to `bv.min_[2]` withouth changing the indentation as:
```cpp
if(vel[2] > 0) bv.max_[2] += vel[2];
else bv.min_[2] += vel[2];
```

File: include/fcl/broadphase/detail/hierarchy_tree-inl.h [hunk 4]
```diff
@@ -1113,7 +1113,7 @@ struct SelectImpl<S, AABB<S>>
...
```
Comment (jslee02) [inline include/fcl/broadphase/detail/hierarchy_tree-inl.h]: and
```cpp
if(vel[2] > 0) bv.max_[2] += vel[2];
else bv.min_[2] += vel[2];
```

**Comment by lunixoid** [comment]
Deleted indentations.

**Comment by chhtz** [comment]
The following should do (essentially) the same, with less code and without branches:

        bv.max_ += vel.cwiseMax(S(0.0));
        bv.min_ += vel.cwiseMin(S(0.0));

Btw: This line looks wrong (unless you somehow overwrote the Eigen constructors):

    Vector3<S> marginv(margin);

This will cast `margin` to `Eigen::Index` and assert that it is 3 (when compiled without `-DNDEBUG`).
You probably wanted to write one of these:

        bv.min_.array() -= margin;
        bv.min_ -= Vector3<S>::Constant(margin);

Make sure to compile your test-suite without `NDEBUG`, if you want to catch these errors.

**Pull request merged**